src/TLuaInterpreter.cpp

```diff
@@ -11466,25 +11495,157 @@ bool TLuaInterpreter::callEventHandler(const QString& function, const TEvent& pE
-                lua_pushnumber(L, pE.mArgumentList.at(i).toDouble());
-                break;
-            case ARGUMENT_TYPE_STRING:
-                lua_pushstring(L, pE.mArgumentList.at(i).toUtf8().constData());
-                break;
-            case ARGUMENT_TYPE_BOOLEAN:
-                lua_pushboolean(L, pE.mArgumentList.at(i).toInt());
-                break;
-            case ARGUMENT_TYPE_NIL:
-                lua_pushnil(L);
-                break;
-            default:
-                qWarning(R"(TLuaInterpreter::callEventHandler("%s", TEvent) ERROR: Unhandled ARGUMENT_TYPE: %i encountered in argument %i.)", function.toUtf8().constData(), pE.mArgumentTypeList.at(i), i);
-                lua_pushnil(L);
+        case ARGUMENT_TYPE_NUMBER:
+            lua_pushnumber(L, pE.mArgumentList.at(i).toDouble());
+            break;
+        case ARGUMENT_TYPE_STRING:
+            lua_pushstring(L, pE.mArgumentList.at(i).toUtf8().constData());
+            break;
+        case ARGUMENT_TYPE_BOOLEAN:
+            lua_pushboolean(L, pE.mArgumentList.at(i).toInt());
+            break;
+        case ARGUMENT_TYPE_NIL:
+            lua_pushnil(L);
+            break;
+        default:
+            qWarning(R"(TLuaInterpreter::callEventHandler("%s", TEvent) ERROR: Unhandled ARGUMENT_TYPE: %i encountered in argument %i.)", function.toUtf8().constData(), pE.mArgumentTypeList.at(i), i);
+            lua_pushnil(L);
         }
     }
 
-    error = lua_pcall(L, pE.mArgumentList.size(), LUA_MULTRET, 0);
+    if (qE != 0) {
+        // Create Lua table with QEvent data if needed
+        switch (qE->type()) {
+        // This means the default argument value was used, so ignore
+        case (QEvent::None):
+            error = lua_pcall(L, pE.mArgumentList.size(), LUA_MULTRET, 0);
+            break;
+        // These are all QMouseEvents
+        case (QEvent::MouseButtonPress):
+        case (QEvent::MouseButtonDblClick):
+        case (QEvent::MouseButtonRelease):
+        case (QEvent::MouseMove): {
+            auto qME = static_cast<const QMouseEvent*>(qE);
+            lua_newtable(L);
+
+            // push button()
+            lua_pushstring(L, mMouseButtons.value(qME->button()).toUtf8().constData());
+            lua_setfield(L, -2, QStringLiteral("button").toUtf8().constData());
+
+            // push buttons()
+            lua_pushstring(L, QStringLiteral("buttons").toUtf8().constData());
+            lua_newtable(L);
+            QMap<Qt::MouseButton, QString>::const_iterator iter = mMouseButtons.constBegin();
+            int counter = 1;
+            while (iter != mMouseButtons.constEnd()) {
+                if (iter.key() & qME->buttons()) {
+                    lua_pushnumber(L, counter);
+                    lua_pushstring(L, iter.value().toUtf8().constData());
+                    lua_settable(L, -3);
+                    counter++;
+                }
+                ++iter;
+            }
+            lua_settable(L, -3);
```

Comment:
This can also be set with the `lua_setfield(L, -2, QStringLiteral("buttons").toUtf8().constData());` function as the table is a value as any other.